### ISSUE TITLE
[Questions] How does Helixer handle Transposable Elements?

### ISSUE THREAD

**Issue by crakshay1** [weberlab-hhu/Helixer#216]
Hey, I'm currently using Helixer to annotate the Zea Mays genome. Given that approximately 85% of this assembly consists of transposable elements (TEs),

_How does Helixer handle TEs without any soft masking? 
How does the current model architecture distinguish between TEs and coding genes? 
Was the training set supplemented with 'hard' examples of TEs to minimize false-positive gene calls in repeat-rich regions?
Does Helixer implement any internal 'masking-by-classification' logic, or does it rely entirely on the learned spatial dependencies of the sequence to bypass repetitive elements?_

Thank you for answering my questions.

**Comment by Artifice120** [comment]
Not a developer, 
Noticed that after running Helixer on the unmasked genome of a fungus with around the same percentage of TE's the gene caller preformed gene calling on both the TE's as well as the genes inflating the gene calls. In the end fine tuning the model with rna-seq alignments improved the gene calls as well as masking the TE and repeats with EDTA. Then searching the remaining genes in EnTAP to label any TE's that remained in the gene calls

**Comment by felicitas215** [comment]
@Artifice120  thank you for answering this question and sharing your experiences.

@crakshay1
Helixer relies on the learned spatial context and implicit masking. It only trained on protein-coding genes, meaning that TEs were labeled as "intergenic" during training, essentially masking them. This has the advantage hat softmasking is not necessary (or even possible) for Helixer. Not relying on softmasking can also prevent genes overlapping TEs getting masked. But, since Helixer was not trained on softmasked genomes, its ability to successfully exclude TEs from the annotation relies on the training data. If the data lacked sufficient information to distinguish TEs and protein-coding genes for a specific species or family, Helixer will annotate the TEs. In these cases hardmasking (which has an effect on Helixer) can be helpful.

**Comment by crakshay1** [comment]
@Artifice120 Thank you for answering my question !
@felicitas215 Thank you for your explanation !

I understood your explanation, however I have another question : Can you tell me what are the labels you used for the training?

**Comment by felicitas215** [comment]
The labels Helixer uses are the four classes "intergenic", "UTR", "CDS" and "intron".
